File: poc-kronos-frontend/src/model/periodoexecucao/FaseExecucao.js
class FaseExecucao {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataInicio = dados ? dados.dataInicio : null;
    this.dataFim = dados ? dados.dataFim : null;
    this.idPeriodoExecucao = dados ? dados.idPeriodoExecucao : null;
    this.tipoPeriodo = dados ? dados.tipoPeriodo : null;
    this.idCurso = dados ? dados.idCurso : null;
    this.idFase = dados ? dados.idFase : null;
    this.siglaFase = dados ? dados.siglaFase : null;
    this.numeroFase = dados ? dados.numeroFase : null;
    this.numeroSubFases = dados ? dados.numeroSubFases : null;
    this.idCurso = dados ? dados.idCurso : null;
  }

  static validations(required, dataFimMenor) {
    return {
      idPeriodoExecucao: { required },
      idFase: { required },
      dataInicio: { required },
      dataFim: { required, dataFimMenor: dataFimMenor("dataInicio") },
    };
  }

  static validationMessages() {
    return {
      dataInicio: {
        required: "Data Início é obrigatório"
      },
      dataFim: {
        required: "Data Final é obrigatória",
        dataFimMenor: "Data Final não pode ser menor que a Data Inicial"
      },
      idPeriodoExecucao: {
        required: "Período Execução é obrigatório"
      },
      idFase: {
        required: "Fase é obrigatória"
      }
    };
  }
}
export default FaseExecucao;

File: src/main/java/br/com/kronos/backend/aplicacao/calendario/FaseExecucao.java

package br.com.kronos.backend.aplicacao.calendario;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class FaseExecucao {
    
    private long id;
    private LocalDate dataInicio;
    private LocalDate dataFim;
    private Long idPeriodoExecucao;
    private Long idFase;

    
 
	@Builder
    public FaseExecucao(long id, LocalDate dataInicio, LocalDate dataFim, Long idPeriodoExecucao, Long idFase) {

        this.id = id;
        this.dataInicio = exigirNaoNulo(dataInicio, "Data de início"); 
        this.dataFim = exigirNaoNulo(dataFim, "Data final"); 
        this.idPeriodoExecucao = exigirNaoNulo(idPeriodoExecucao, "Período (Ano) de execução"); 
        this.idFase = exigirNaoNulo(idFase, "Fase (Semestre)"); 
	}
}